### PR TITLE
[WL cleanup] #trivial simplify error handling

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_

- remove no-op checks
- remove unnecessary try-catch
- consolidate unauthorized error check

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: packages/wallet-sdk/src/core/error/serialize.ts [hunk 1]
```diff
@@ -35,10 +35,17 @@ function getErrorObject(error: string | Web3Response | unknown) {
       code: standardErrorCodes.rpc.internal,
     };
   } else if (isErrorResponse(error)) {
+    const message = error.errorMessage;
+    const code =
+      error.errorCode ??
+      (message.match(/(denied|rejected)/i)
+        ? standardErrorCodes.provider.userRejectedRequest
+        : undefined);
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/error/serialize.ts:43]: image

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 2]
```diff
@@ -358,16 +345,6 @@ export class WalletLinkSigner implements Signer {
...
 
-  private _isAuthorized(): boolean {
-    return this._addresses.length > 0;
-  }
-
-  private _requireAuthorization(): void {
-    if (!this._isAuthorized()) {
-      throw standardErrors.provider.unauthorized({});
-    }
-  }
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:369]: BS

File: packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts [hunk 3]
```diff
@@ -451,7 +407,6 @@ export class WalletLinkSigner implements Signer {
   }
 
   private _personal_sign(params: RequestParam) {
-    this._requireAuthorization();
```
Comment (bangtoven) [inline packages/wallet-sdk/src/sign/walletlink/WalletLinkSigner.ts:454]: no need to call `_requireAuthorization`
it's handled on the provider level before hitting WLSigner

**Review by nateReiners — APPROVED** [review]

**Pull request merged**